File: fixtures/scale/python/large/cipherlab/checks/check_djbhash.py
from cipherlab.ciphers.djbhash import djbhash_digest
from cipherlab.specs.djbhash_spec import djbhash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_djbhash():
    spec = djbhash_spec()
    for text in corpus_hash():
        first = djbhash_digest(text)
        second = djbhash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"

File: fixtures/scale/ruby/large/lib/check_djbhash.rb
require_relative 'djbhash_digest'
require_relative 'djbhash_digest_pair'
require_relative 'corpus_hash'

def check_djbhash
  corpus_hash.each do |text|
    first = djbhash_digest(text)
    second = djbhash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end
